### PR TITLE
Add annotations for Prometheus service discovery

### PR DESCRIPTION
**Pull request by santinoncs**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
/kind feature
> /kind flake

**What this PR does / why we need it**:

Kube-dns pods are exporting /metrics but there is no way prometheus can scrape these metrics.
So, we set these annotations to be able to use service discovery feature of Prometheus.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cluster/addons/dns/kube-dns/kube-dns.yaml.base [hunk 1]
```diff
@@ -84,6 +84,9 @@ spec:
       annotations:
         scheduler.alpha.kubernetes.io/critical-pod: ''
         seccomp.security.alpha.kubernetes.io/pod: 'docker/default'
+        prometheus.io/path: /metrics
```
Comment (MrHohn) [inline cluster/addons/dns/kube-dns/kube-dns.yaml.base]: Isn't `metrics` the default so we can omit this annotation?

**Review by MrHohn — APPROVED** [review]
This seems reasonable as this is what we do for [coredns manifests](https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/dns/coredns/coredns.yaml.base)?
/approve
/cc @bowei @prameshj

File: cluster/addons/dns/kube-dns/kube-dns.yaml.base (hunk 1 above)
Comment (santinoncs) [reply cluster/addons/dns/kube-dns/kube-dns.yaml.base]: fixed!

**Pull request merged**